src/Avalonia.Controls/CalendarDatePicker/CalendarDatePicker.cs

```diff
@@ -554,17 +561,21 @@ private void Calendar_DisplayDateChanged(object? sender, CalendarDateChangedEven
         {
             if (e.AddedDate != this.DisplayDate)
             {
-                SetValue(DisplayDateProperty, (DateTime) e.AddedDate!);
+                SetValue(DisplayDateProperty, (DateTimeOffset)e.AddedDate!);
```

Comment:
I considered here (and in the below Calendar_SelectedDatesChanged method) whether the `DateTime` from Calendar should be converted to a `DateTimeOffset` as follows `new DateTimeOffset(dateTime.Ticks, TimeSpan.Zero)`.

Doing this would ensure the time zone component is always zeroed out -- otherwise the DateTimeOffset will convert Kind.Unspecified into the local computer's time zone. There was concern initially that this automatic time zone conversion would change the selected date forward or backward up to 1 day in the worse case.

However, this shouldn't actually happen since the `DateTime` coming from Calendar always has the time zeroed to 12:00:00am/00:00:00. I think any automatic time zone adjustment would therefore always be in the same day. And we actually want the `SelectedDate` in the `CalendarDatePicker` to return the local time zone.